examples/manual_encrypted_verify.py

```diff
@@ -0,0 +1,199 @@
+
+    Arguments:
+        resp {LoginResponse} -- the successful client login response.
+    """
+    with open(SESSION_DETAILS_FILE, "w") as f:
+        json.dump({
+            "access_token": resp.access_token,
+            "device_id": resp.device_id,
+            "user_id": resp.user_id
+        }, f)
+
+
+async def main() -> None:
+    """A very simple encrypted nio application.
+    """
+    # If the store location doesn't exist, we'll make it
+    if not os.path.isdir(STORE_FOLDER):
+        os.mkdir(STORE_FOLDER)
+
+    # By setting `store_sync_tokens` to true, we'll save sync tokens to our
+    # store every time we sync, thereby preventing reading old, previously read
+    # events on each new sync.
+    # For more info, check out https://matrix-nio.readthedocs.io/en/latest/nio.html#asyncclient
+    config = ClientConfig(store_sync_tokens=True)
+    client = AsyncClient(
+        ALICE_HOMESERVER,
+        ALICE_USER_ID,
+        store_path=STORE_FOLDER,
+        config=config
+    )
+
+    # restore the previous session if we can, otherwise login
+    if os.path.exists(SESSION_DETAILS_FILE) and os.path.isfile(SESSION_DETAILS_FILE):
+        try:
+            with open(SESSION_DETAILS_FILE, "r") as f:
+                config = json.load(f)
+                client.access_token = config['access_token']
+                client.user_id = config['user_id']
+                client.device_id = config['device_id']
+
+                # This loads our verified/blacklisted devices and our keys
+                client.load_store()
+                print(f"Logged in using stored credentials: {client.user_id} on {client.device_id}")
+
+        except IOError as err:
+            print(f"Couldn't load session from file. Logging in. Error: {err}")
+        except json.JSONDecodeError as err:
+            print(f"Couldn't read JSON file; overwriting")
+
+    if not client.user_id or not client.access_token or not client.device_id:
+        resp = await client.login(ALICE_PASSWORD)
+        if isinstance(resp, LoginResponse):
+            print("Logged in using a password; saving details to disk")
+            write_details_to_disk(resp)
+        else:
+            print(f"Failed to log in: {resp}")
+            sys.exit(1)
+
+    # If your room isn't public, you'll need to invite this user before
+    # it joins the room
+    await client.join(ROOM_ID)
+    await client.sync(full_state=True)
+
+    room = client.rooms[ROOM_ID]
+    print(f"Room {room.name} is encrypted: {room.encrypted}" )
+
+    # nio will transparently encrypt the message for us but other users cannot
+    # read it as we have not shared our keys and haven't received the keys of
+    # anyone else.
+    await client.room_send(
+        room_id=ROOM_ID,
+        message_type="m.room.message",
+        content = {
+            "msgtype": "m.text",
+            "body": "An encrypted message you cannot read"
+        }
+    )
+
+    if (client.should_upload_keys):
```

Comment:
Heh, right, that's what `sync_forever()` does. If you want to show how it's done manually, yeah we can do this without the asyncio task.

Reply:
Ahhh; I was having problems because this example doesn't use `sync_forever()` so I added it in, but maybe I'll change this around so we use sync_forever instead, and as a task